extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/logging/RequestLoggerRecorder.java

```diff
@@ -0,0 +1,20 @@
+package io.quarkus.vertx.http.runtime.logging;
+
+import io.quarkus.runtime.annotations.Recorder;
+import io.vertx.core.Handler;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.LoggerHandler;
+
+@Recorder
+public class RequestLoggerRecorder {
+
+    private static final RequestLogMessageFormatter REQUEST_LOG_MESSAGE_FORMATTER = new RequestLogMessageFormatter();
+
+    public static final VertxLoggerFormatMapper VERTX_LOGGER_FORMAT_MAPPER = new VertxLoggerFormatMapper();
+
+    public Handler<RoutingContext> handler(RequestLoggerFormat format) {
+        return VERTX_LOGGER_FORMAT_MAPPER.map(format).<Handler<RoutingContext>> map(LoggerHandler::create)
```

Comment:
Avoid using lambdas in recorder classes

Reply:
And `Optional` too, just check for the good old `null` :)

Reply:
Just for my information, why using lambdas and Optional should be avoided? We should avoid it only in recorder classes or in general?

Reply:
Because of performance issues if i remember correctly. We usually avoid using them in our extensions

Reply:
Optional is ok, its just that it is not widely used in the code base outside of config, so it just keeps things consistent. 

Lambdas use up around 1k of metaspace each, that is memory that will never be released and hang around for the life of the application, while an 'if' statement is basically free. We do use some in the build time code (-deployment) where we don't care about memory as much, but in the runtime code we don't want them, especially in cases like this where the alternative is an if statement. They are also slower as the JVM has to define a new lambda.